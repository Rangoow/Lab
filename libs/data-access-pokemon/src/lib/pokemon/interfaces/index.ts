export * from './pokemon-lite.interface';
export * from './pokemon-lite-paginated.interface';
