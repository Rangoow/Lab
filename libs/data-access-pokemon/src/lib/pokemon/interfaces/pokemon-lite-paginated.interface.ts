import { IPokemonLite } from './pokemon-lite.interface';

export interface IPokemonLitePaginated {
  readonly count: number;
  readonly next: string;
  readonly previous: number;
  readonly results: IPokemonLite[];
}
