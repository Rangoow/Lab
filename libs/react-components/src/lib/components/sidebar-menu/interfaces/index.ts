export * from './sidebar-menu-element.interface';
