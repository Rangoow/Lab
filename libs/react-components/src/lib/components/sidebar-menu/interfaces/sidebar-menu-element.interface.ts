import { VitamixId } from '@vtmn/icons/dist/vitamix/font/vitamix';

export interface ISidebarMenuElement {
  readonly key: string;
  readonly text: string;
  readonly title: string;
  readonly icon: VitamixId;
  readonly redirectPath: string;
}
