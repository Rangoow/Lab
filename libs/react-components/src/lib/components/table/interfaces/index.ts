export * from './table-demo.interface';
export * from './progress-bar.interface';
