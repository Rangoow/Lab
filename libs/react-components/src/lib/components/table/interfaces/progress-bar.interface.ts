export interface IUiProgressBar {
  readonly current: number;
  readonly max: number;
}
