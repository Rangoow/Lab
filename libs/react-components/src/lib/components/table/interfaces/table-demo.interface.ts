import { EDemoStatusTag } from '../enums';
import { IUiProgressBar } from './progress-bar.interface';

export interface IUiTableDemo {
  string: string;
  number: number;
  image: string;
  date: string;
  boolean: boolean;
  status: EDemoStatusTag;
  progressBar: IUiProgressBar;
}
