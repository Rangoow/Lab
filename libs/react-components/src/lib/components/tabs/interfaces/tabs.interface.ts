import { VitamixId } from '@vtmn/icons/dist/vitamix/font/vitamix';

export interface ITabs {
  readonly id: string;
  readonly titleTranslationKey: string;
  readonly path: string;
  readonly icon?: VitamixId;
  readonly badgeValue?: number;
}
