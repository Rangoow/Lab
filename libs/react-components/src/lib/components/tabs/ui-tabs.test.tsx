import { randUuid, randWord } from '@ngneat/falso';
import { RenderResult, screen } from '@testing-library/react';
import { UserEvent, userEvent } from '@testing-library/user-event';
import * as router from 'react-router';
import { Mock, MockInstance } from 'vitest';
import { customRender } from '../../tests/custom-render';
import { UiTabs } from './ui-tabs.component';
import { ITabs } from './interfaces';

const user: UserEvent = userEvent.setup();
const mockedUseNavigate: Mock = vi.fn();

describe('uiTabs', (): void => {
  let tabs: ITabs[];
  let spyOnNavigate: MockInstance;

  describe('snapshot testing', (): void => {
    beforeEach((): void => {
      tabs = [
        { titleTranslationKey: 'tab1', path: 'tab_1', id: 'tab_1_id' },
        { titleTranslationKey: 'tab2', path: 'tab_2', id: 'tab_2_id', icon: 'user-line', badgeValue: 10 },
      ];
    });

    it(`should render correctly and match the snapshot`, async (): Promise<void> => {
      expect.assertions(1);

      const result: RenderResult = await customRender(<UiTabs tabs={tabs} />);

      expect(result).toMatchSnapshot();
    });
  });

  describe('component testing', (): void => {
    const tabTitle1: string = randWord();
    const tabTitle2: string = randWord();

    beforeEach((): void => {
      tabs = [
        { titleTranslationKey: tabTitle1, path: 'tab_1', id: randUuid() },
        {
          titleTranslationKey: tabTitle2,
          path: 'tab_2',
          id: randUuid(),
          icon: 'user-line',
        },
      ];
      spyOnNavigate = vi.spyOn(router, `useNavigate`).mockImplementation(() => mockedUseNavigate);
    });

    it(`should display a tab list with the correct number of tabs`, async (): Promise<void> => {
      expect.assertions(2);

      await customRender(<UiTabs tabs={tabs} />);
      const tabListElement: HTMLElement = screen.getByRole('tablist');
      const tabElements: HTMLElement[] = screen.getAllByRole('tab');

      expect(tabListElement).toBeVisible();
      expect(tabElements).toHaveLength(tabs.length);
    });

    it('should navigate when the user click on a tab', async (): Promise<void> => {
      expect.assertions(1);

      await customRender(<UiTabs tabs={tabs} />);
      const tab2Element: HTMLElement = screen.getByRole('button', { name: tabTitle2 });

      await user.click(tab2Element);

      expect(spyOnNavigate).toHaveBeenCalledOnce();
    });
  });
});
