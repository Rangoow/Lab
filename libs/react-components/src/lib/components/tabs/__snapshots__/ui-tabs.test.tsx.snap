// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiTabs > snapshot testing > should render correctly and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul
          align="start"
          class="vtmn-tabs vtmn-tabs_align--start vtmn-tabs_size--medium"
          role="tablist"
        >
          <li
            role="tab"
          >
            <button
              aria-selected="false"
              title="tab1"
            >
              tab1
            </button>
          </li>
          <li
            role="tab"
          >
            <button
              aria-selected="false"
              title="tab2"
            >
              <span
                aria-hidden="true"
                class="vtmx-user-line "
              />
              tab2
              <span
                class="vtmn-badge vtmn-badge_variant--default "
              >
                10
              </span>
            </button>
          </li>
        </ul>
        <div
          aria-orientation="horizontal"
          class="vtmn-divider vtmn-divider_orientation--horizontal"
          role="separator"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul
        align="start"
        class="vtmn-tabs vtmn-tabs_align--start vtmn-tabs_size--medium"
        role="tablist"
      >
        <li
          role="tab"
        >
          <button
            aria-selected="false"
            title="tab1"
          >
            tab1
          </button>
        </li>
        <li
          role="tab"
        >
          <button
            aria-selected="false"
            title="tab2"
          >
            <span
              aria-hidden="true"
              class="vtmx-user-line "
            />
            tab2
            <span
              class="vtmn-badge vtmn-badge_variant--default "
            >
              10
            </span>
          </button>
        </li>
      </ul>
      <div
        aria-orientation="horizontal"
        class="vtmn-divider vtmn-divider_orientation--horizontal"
        role="separator"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`uiTabs > snapshot testing > should render correctly and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul
          align="start"
          class="vtmn-tabs vtmn-tabs_align--start vtmn-tabs_size--medium"
          role="tablist"
        >
          <li
            role="tab"
          >
            <button
              aria-selected="false"
              title="tab1"
            >
              tab1
            </button>
          </li>
          <li
            role="tab"
          >
            <button
              aria-selected="false"
              title="tab2"
            >
              <span
                aria-hidden="true"
                class="vtmx-user-line "
              />
              tab2
              <span
                class="vtmn-badge vtmn-badge_variant--default "
              >
                10
              </span>
            </button>
          </li>
        </ul>
        <div
          aria-orientation="horizontal"
          class="vtmn-divider vtmn-divider_orientation--horizontal"
          role="separator"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul
        align="start"
        class="vtmn-tabs vtmn-tabs_align--start vtmn-tabs_size--medium"
        role="tablist"
      >
        <li
          role="tab"
        >
          <button
            aria-selected="false"
            title="tab1"
          >
            tab1
          </button>
        </li>
        <li
          role="tab"
        >
          <button
            aria-selected="false"
            title="tab2"
          >
            <span
              aria-hidden="true"
              class="vtmx-user-line "
            />
            tab2
            <span
              class="vtmn-badge vtmn-badge_variant--default "
            >
              10
            </span>
          </button>
        </li>
      </ul>
      <div
        aria-orientation="horizontal"
        class="vtmn-divider vtmn-divider_orientation--horizontal"
        role="separator"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
